src/sentry/search/snuba/executors.py

```diff
@@ -1215,19 +1213,39 @@ def get_basic_event_snuba_condition(
         if not isinstance(item, list):
             raw_conditions = [raw_conditions]
 
+        query_builder = self.def_get_query_builder(joined_entity)
+        query_builder.default_filter_converter(search_filter)
+
+        conver_value_to_date = False
         output_conditions = []
         for item in raw_conditions:
-            column_name = item[0]
-            # do some name mapping for snuba
-            column_name = column_name.replace("stack.", "stacktrace.")
-            if ATTR_CHOICES.get(column_name) is not None:
-                raw_column = ATTR_CHOICES.get(column_name)
-                column_name = raw_column.value.event_name
-
-            column = Column(column_name, joined_entity)
+            lhs = item[0]
+            if isinstance(lhs, str):
+                raw_column = map_field_name_from_format_search_filter(lhs)
+                lhs = query_builder.resolve_column(raw_column)
+            else:
+                # need to convert dates to timestamps
+                if lhs[1][0] == "date":
+                    conver_value_to_date = True
+                # right now we are assuming lhs looks like ['isNull', ['user']]
+                # if there are more complex expressions we will need to handle them
+                raw_column = map_field_name_from_format_search_filter(lhs[1][0])
+                rhs = [query_builder.resolve_column(raw_column)]
+                if len(lhs[1]) > 1:
+                    rhs.append(lhs[1][1])
+                lhs = Function(lhs[0], rhs)
```

Comment:
why do we have to do this? looks weird

Reply:
@JoshFerge we have to deal with nested arrays

Reply:
got it, mind adding an example in the comment above?

Reply:
@JoshFerge added